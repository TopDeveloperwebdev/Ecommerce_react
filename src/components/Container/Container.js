import React from 'react';
import PropTypes from 'prop-types';

import style from './Container.sass';

const Container = ({ children, width, height, alignItems, justifyContent, styles }) => (
  <div className={style.container} style={{ width, height, justifyContent, alignItems, ...styles }} >{children}</div>
);

Container.propTypes = {
  children: PropTypes.node,
  width: PropTypes.string,
  height: PropTypes.string,
  alignItems: PropTypes.string,
  justifyContent: PropTypes.string,
  styles: PropTypes.objectOf(PropTypes.string),
};

Container.defaultProps = {
  children: '',
  width: '100%',
  height: 'auto',
  alignItems: 'flex-start',
  justifyContent: '',
  styles: {},
};

export default Container;
