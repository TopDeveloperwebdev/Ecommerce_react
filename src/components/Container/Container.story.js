import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo, setDefaults } from '@storybook/addon-info';

import Item from 'components/Item';
import Box from 'components/Box';
import Container from './Container';

// addon-info
setDefaults({
  source: true,
});

storiesOf('Container', module)
  .add('Default', withInfo()(() => (
    <Container justifyContent="center" alignItems="center" height="500px" styles={{ backgroundColor: '#f1f1f1', flexDirection: 'row' }}>
      <Item>
        <Box>
          Box inside a container flex
        </Box>
      </Item>
    </Container>
  )))
  .add('With 3 items', withInfo()(() => (
    <Container justifyContent="center" alignItems="center" height="500px" styles={{ backgroundColor: '#f1f1f1', flexDirection: 'row' }}>
      <Item flexGrow="1">
        <Box>
          Box inside a container flex
        </Box>
      </Item>
      <Item flexGrow="1">
        <Box>
          Box inside a container flex
        </Box>
      </Item>
      <Item flexGrow="1">
        <Box>
          Box inside a container flex
        </Box>
      </Item>
    </Container>
  )));
