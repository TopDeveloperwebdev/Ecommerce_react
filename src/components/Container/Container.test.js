import React from 'react';
import uid from 'uid';
import { shallow, mount } from 'enzyme';

import Container from './Container';

test('Should component render text', () => {
  const text = 'text';

  const wrapper = mount(<Container>{text}</Container>);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.container').text()).toEqual(text);
});
