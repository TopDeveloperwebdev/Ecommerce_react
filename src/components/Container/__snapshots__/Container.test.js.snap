// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should component render text 1`] = `
<Container
  alignItems="flex-start"
  height="auto"
  justifyContent=""
  styles={Object {}}
  width="100%"
>
  <div
    className="container"
    style={
      Object {
        "alignItems": "flex-start",
        "height": "auto",
        "justifyContent": "",
        "width": "100%",
      }
    }
  >
    text
  </div>
</Container>
`;
